dev/tools/bin/engine_hash.dart

```diff
@@ -0,0 +1,162 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: always_specify_types
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:crypto/crypto.dart';
+
+enum GitRevisionStrategy {
+  mergeBase,
+  head,
+}
+
+final _hashRegex = RegExp(r'^([a-fA-F0-9]+)');
+
+final parser = ArgParser()
```

Comment:
How picky are we with conforming to the framework style guide for tooling like this? I personally prefer omitting types on the LHS when the type is obvious, but I know that's not typically what we do for framework / tool code.